genai-perf/genai_perf/config/generate/perf_analyzer_config.py

```diff
@@ -311,6 +317,37 @@ def get_obj_args(self) -> Namespace:
 
         return obj_args
 
+    def get_inference_type(self) -> InferenceType:
+        """
+        Returns the type of inferencing: concurrency or request-rate
+        """
+        cmd = self.create_command()
+        if "--concurrency-range" in cmd:
+            return InferenceType.CONCURRENCY
+        elif "--request-rate-range" in cmd:
+            return InferenceType.REQUEST_RATE
+        else:
+            raise GenAIPerfException(
+                f"An inference type (either concurrency or request-rate) was not found."
+            )
+
+    def get_inference_value(self) -> int:
+        """
+        Returns the value that we are inferencing
+        """
+        infer_type = self.get_inference_type()
+        infer_cmd_option = (
+            "--concurrency-range"
+            if infer_type == InferenceType.CONCURRENCY
+            else "--request-rate-range"
+        )
+
+        cmd = self.create_command()
+        infer_value_index = cmd.index(infer_cmd_option) + 1
```

Comment:
This feels brittle.
Would a dictionary be safer as the code base grows?

Reply:
Then I think we have to come up with a better solution at some point but for now, we move forward. Nice work.